packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -235,4 +235,50 @@ export class Span implements api.Span, ReadableSpan {
     }
     return this._ended;
   }
+
+  // Utility function to truncate given value within size
+  // for value type of string, will truncate to given limit
+  // for type of non-string, will return same value
+  private _truncateToLimitUtil(value: string, limit: number): string {
+    if (value.length <= limit) {
+      return value;
+    }
+    return value.substr(0, limit);
+  }
+
+  /**
+   * If the given attribute value is of type string and has more characters than given {@code attributeValueLengthLimit} then
+   * return string with trucated to {@code attributeValueLengthLimit} characters
+   *
+   * If the given attribute value is array of strings then
+   * return new array of strings with each element truncated to {@code attributeValueLengthLimit} characters
+   *
+   * Otherwise return same Attribute {@code value}
+   *
+   * @param value Attribute value
+   * @returns truncated attribute value if required, otherwise same value
+   */
+  private _truncateToSize(value?: SpanAttributeValue): SpanAttributeValue | undefined {
+    const limit = this._spanLimits.attributeValueLengthLimit;
+    // Check limit
+    // undefined limit means do not truncate
+    if (typeof limit !== 'number' || limit <= 0) {
```

Comment:
Could we move this check to evaluate only once during SDK init/config update instead on every setAttributed?

Reply:
Done!